modApiServer/test/org/aion/api/server/pb/ApiAion0Tests.java

```diff
@@ -0,0 +1,1304 @@
+ *     Aion foundation.
+ */
+
+package org.aion.api.server.pb;
+
+import com.google.protobuf.ByteString;
+import org.aion.api.server.ApiUtil;
+import org.aion.base.type.Address;
+import org.aion.base.util.TypeConverter;
+import org.aion.crypto.ed25519.ECKeyEd25519;
+import org.aion.equihash.EquihashMiner;
+import org.aion.mcf.account.AccountManager;
+import org.aion.mcf.account.Keystore;
+import org.aion.zero.impl.Version;
+import org.aion.zero.impl.blockchain.AionImpl;
+import org.aion.zero.impl.config.CfgAion;
+import org.aion.zero.impl.db.AionRepositoryImpl;
+import org.aion.zero.impl.types.AionBlock;
+import org.aion.zero.impl.types.AionBlockSummary;
+import org.aion.zero.types.AionTransaction;
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+public class ApiAion0Tests {
+
+    private byte[] msg, socketId, hash, rsp;
+
+    private ApiAion0 api;
+
+
+    private static final int MSG_HASH_LEN = 8;
+    private static final int RSP_HEADER_NOHASH_LEN = 3;
+    private static final int REQ_HEADER_NOHASH_LEN = 4;
+    private static final int RSP_HEADER_LEN = RSP_HEADER_NOHASH_LEN + MSG_HASH_LEN;
+
+    public ApiAion0Tests() {
+        msg = "test message".getBytes();
+        socketId = RandomUtils.nextBytes(5);
+        hash = RandomUtils.nextBytes(ApiUtil.HASH_LEN);
+        rsp = null;
+    }
+
+    private byte[] stripHeader(byte[] rsp) {
+        boolean hasHash = (rsp[2] == 1);
+        int bodyLen = rsp.length - (hasHash ? RSP_HEADER_LEN : RSP_HEADER_NOHASH_LEN);
+
+        if (hasHash) {
+            return Arrays.copyOfRange(rsp, RSP_HEADER_LEN, RSP_HEADER_LEN + bodyLen);
+        } else {
+            return Arrays.copyOfRange(rsp, RSP_HEADER_NOHASH_LEN, RSP_HEADER_NOHASH_LEN + bodyLen);
+        }
+    }
+
+    private byte[] sendRequest(int s, int f) {
+        byte[] request = ByteBuffer.allocate(msg.length + REQ_HEADER_NOHASH_LEN + hash.length).put(api.getApiVersion())
+                .put((byte) s)
+                .put((byte) f)
+                .put((byte) 1)
+                .put(hash)
+                .put(msg).array();
+
+        return api.process(request, socketId);
+    }
+
+    private byte[] sendRequest(int s, int f, byte[] reqBody) {
+        byte[] request = ByteBuffer.allocate(reqBody.length + REQ_HEADER_NOHASH_LEN + hash.length).put(api.getApiVersion())
+                .put((byte) s)
+                .put((byte) f)
+                .put((byte) 1)
+                .put(hash)
+                .put(reqBody)
+                .array();
+
+        return api.process(request, socketId);
+    }
+
+    private void setup() {
+        api = new ApiAion0(AionImpl.inst());
+    }
+
+    private void tearDown() {
+        api.shutDown();
+        rsp = null;
+    }
+
+    @Test
+    public void testHeartBeatMsg() {
+        setup();
```

Comment:
With JUnit you can actually just put a `@Before` annotation above your setup method and `@After` above your tearDown and they will be called before and after every single test case respectively, and then you can remove the explicit calls to setup and teardown :)